### PR TITLE
[bitnami/postgresql]  postgresql backup container adds resources parameter

### LINKED ISSUE #23938: [bitnami/postgresql] postgresql backup does not have a resources parameter
### Name and Version

bitnami/postgresql 14.2.3

### What is the problem this feature will solve?

On some multi-tenant clusters a `resources` definition is mandatory for any container to start. Otherwise, you get an error like this : 

```
❯ k describe jobs.batch postgresql-primary-pgdumpall-28482480

Name:             postgresql-primary-pgdumpall-28482480
Namespace:        xxxxxxxx
Selector:         controller-uid=b863d909-887a-43f6-bc1c-b5b08238d7c1
Labels:           app.kubernetes.io/component=pg_dumpall
                  app.kubernetes.io/instance=postgresql
                  app.kubernetes.io/managed-by=Helm
                  app.kubernetes.io/name=postgresql
                  app.kubernetes.io/version=16.2.0
                  controller-uid=b863d909-887a-43f6-bc1c-b5b08238d7c1
                  helm.sh/chart=postgresql-14.0.5
                  job-name=postgresql-primary-pgdumpall-28482480
Annotations:      <none>
Controlled By:    CronJob/postgresql-primary-pgdumpall
Parallelism:      1
Completions:      1
Completion Mode:  NonIndexed
Pods Statuses:    0 Active / 0 Succeeded / 0 Failed
Pod Template:
  Labels:  app.kubernetes.io/component=pg_dumpall
           app.kubernetes.io/instance=postgresql
           app.kubernetes.io/managed-by=Helm
           app.kubernetes.io/name=postgresql
           app.kubernetes.io/version=16.2.0
           controller-uid=b863d909-887a-43f6-bc1c-b5b08238d7c1
           helm.sh/chart=postgresql-14.0.5
           job-name=postgresql-primary-pgdumpall-28482480
  Containers:
   postgresql-primary-pgdumpall:
    Image:           xxxxxxxxx/bitnami/postgresql:16.2.0-debian-11-r1
    Port:            <none>
    Host Port:       <none>
    SeccompProfile:  RuntimeDefault
    Command:
      /bin/sh
      -c
      pg_dumpall --clean --if-exists --load-via-partition-root --quote-all-identifiers --no-password --file=${PGDUMP_DIR}/pg_dumpall-$(date '+%Y-%m-%d-%H-%M').pgdump
    Environment:
      PGUSER:      postgres
      PGPASSWORD:  <set to the key 'postgres-password' in secret 'postgresql'>  Optional: false
      PGHOST:      postgresql-primary
      PGPORT:      5432
      PGDUMP_DIR:  /backup/pgdump
    Mounts:
      /backup/pgdump from datadir (rw)
  Volumes:
   datadir:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  postgresql-primary-pgdumpall
    ReadOnly:   false
Events:
  Type     Reason        Age                   From            Message
  ----     ------        ----                  ----            -------
  Warning  FailedCreate  5m11s (x12 over 70m)  job-controller  (combined from similar events): Error creating: pods "postgresql-primary-pgdumpall-28482480-dzpjr" is forbidden: failed quota: kaas-compute-resources-netbox-services: must specify limits.cpu,limits.memory
```

In this context, backup can't be used because we can't set `resources` in values.yaml configuration. 

### What is the feature you are proposing to solve the problem?

Add a resources block in [backup/cronjob.yaml](https://github.com/bitnami/charts/blob/main/bitnami/postgresql/templates/backup/cronjob.yaml)

example of how it's done [in primary stateful set](https://github.com/bitnami/charts/blob/main/bitnami/postgresql/templates/primary/statefulset.yaml#L93-L97)

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by opencmit2**
### Description of the change

postgresql backup container adds resources parameter
### Applicable issues

- fixes #23938 

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/postgresql/templates/backup/cronjob.yaml [hunk 1]
```diff
@@ -95,6 +95,9 @@ spec:
             {{- if .Values.backup.cronjob.containerSecurityContext.enabled }}
             securityContext: {{- omit .Values.backup.cronjob.containerSecurityContext "enabled" | toYaml | nindent 14 }}
             {{- end }}
+            {{- if .Values.backup.cronjob.resources }}
```
Comment (javsalgar) [inline bitnami/postgresql/templates/backup/cronjob.yaml:98]: Could you add the resourcesPreset section just like in the rest of the objects?

Comment (opencmit2) [reply bitnami/postgresql/templates/backup/cronjob.yaml:98]: Is this the format?
```yaml
          {{- if .Values.readReplicas.resources }}
          resources: {{- toYaml .Values.readReplicas.resources | nindent 12 }}
          {{- else if ne .Values.readReplicas.resourcesPreset "none" }}
          resources: {{- include "common.resources.preset" (dict "type" .Values.readReplicas.resourcesPreset) | nindent 12 }}
          {{- end }}
```

**Comment by opencmit2** [comment]
@javsalgar 
continue to review？

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**